src/com/amplitude/api/AmplitudeCallbacks.java

```diff
@@ -0,0 +1,65 @@
+package com.amplitude.api;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+import android.util.Log;
+
+public class AmplitudeCallbacks implements Application.ActivityLifecycleCallbacks {
+
+    public static final String TAG = "com.amplitude.api.AmplitudeCallbacks";
+    private static final String NULLMSG = "Need to initialize AmplitudeCallbacks with AmplitudeClient instance";
+
+    private AmplitudeClient clientInstance = null;
+
+    public AmplitudeCallbacks(AmplitudeClient clientInstance) {
+        if (clientInstance == null) {
+            Log.e(TAG, NULLMSG);
+            return;
+        }
+
+        this.clientInstance = clientInstance;
+        clientInstance.useAccurateTracking(true);
+    }
+
+    @Override
+    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {}
+
+    @Override
+    public void onActivityDestroyed(Activity activity) {}
+
+    @Override
+    public void onActivityPaused(Activity activity) {
+        if (clientInstance == null) {
+            Log.e(TAG, NULLMSG);
+            return;
+        }
+
+        clientInstance.refreshSessionTime(getCurrentTimeMillis());
```

Comment:
I feel like this class should just delegate to a clientInstance.onExitForeground(...) / clientInstance.onEnterForeground(...) and have these two lines in those methods